plugins/browser-plugin-focalmeter/src/index.ts

```diff
@@ -0,0 +1,156 @@
+  hasLocalStorage,
+  BrowserTracker,
+  attemptWriteLocalStorage,
+} from '@snowplow/browser-tracker-core';
+import { Logger, Payload } from '@snowplow/tracker-core';
+
+/** FocalMeter plugin configuration */
+export interface FocalMeterConfiguration {
+  /** URL of the Kantar endpoint to send the requests to (including protocol) */
+  kantarEndpoint: string;
+  /** Whether to store information about the last submitted user ID in local storage to prevent sending it again on next load (defaults not to use local storage) */
+  useLocalStorage?: boolean;
+}
+
+const _trackers: Record<string, BrowserTracker> = {};
+const _configurations: Record<string, FocalMeterConfiguration> = {};
+
+/**
+ * The FocalMeter Plugin
+ *
+ * The plugin sends requests with the domain user ID to a Kantar endpoint used with the FocalMeter system.
+ * A request is made when the first event with a new user ID is tracked.
+ *
+ * Call `enableFocalMeterIntegration()` to enable the integration with given configuration.
+ */
+export function FocalMeterPlugin(): BrowserPlugin {
+  let LOG: Logger;
+  let lastUserId: string | undefined | null;
+  let trackerId: string;
+
+  return {
+    activateBrowserPlugin: (tracker: BrowserTracker) => {
+      trackerId = tracker.id;
+      _trackers[tracker.id] = tracker;
+    },
+
+    logger: (logger: Logger) => {
+      LOG = logger;
+    },
+
+    afterTrack: (payload: Payload) => {
+      if (!_configurations[trackerId]) {
+        LOG.error('FocalMeter integration not enabled');
+        return;
+      }
+
+      let newUserId = payload['duid'] as string;
+      let { kantarEndpoint, useLocalStorage } = _configurations[trackerId];
+
+      if (!lastUserId && useLocalStorage && hasLocalStorage()) {
+        let key = getLocalStorageKey(trackerId);
+        lastUserId = attemptGetLocalStorage(key);
+      }
+
+      if (newUserId && newUserId != lastUserId) {
+        lastUserId = newUserId;
+
+        sendRequest(kantarEndpoint, newUserId, LOG, () => {
+          // only write in local storage if the request succeeded
+          if (useLocalStorage && hasLocalStorage()) {
+            let key = getLocalStorageKey(trackerId);
+            attemptWriteLocalStorage(key, newUserId);
+          }
+        });
+      }
+    },
+  };
+}
+
+/**
+ * Enables the integration with Kantar FocalMeter.
+ *
+ * @param configuration - Configuration with the URL endpoint to send requests to
+ * @param trackers - The tracker identifiers which should have the context enabled
+ */
+export function enableFocalMeterIntegration(
+  configuration: FocalMeterConfiguration,
+  trackers: Array<string> = Object.keys(_trackers)
+): void {
+  for (const id of trackers) {
+    if (_trackers[id]) {
+      _configurations[id] = configuration;
+    }
+  }
+}
+
+function getLocalStorageKey(trackerId: string): string {
+  return `sp-fclmtr-${trackerId}`;
+}
+
+function sendRequest(url: string, userId: string, LOG: Logger, successCallback: () => void): void {
```

Comment:
At some point this should be extracted _(just leaving the comment here as a thought moving forward)_